tests/cases/py_client/lib_test.py

```diff
@@ -144,10 +173,10 @@ def testRestCore(self):
         flag = False
         try:
             self.client.getResource('user/badId')
-        except girder_client.HttpError as e:
-            self.assertEqual(e.status, 400)
-            self.assertEqual(e.method, 'GET')
-            resp = json.loads(e.responseText)
+        except requests.HTTPError as e:
+            self.assertEqual(e.response.status_code, 400)
+            self.assertEqual(e.request.method, 'GET')
```

Comment:
It's a little tricky, but `requests.Response` actually holds a reference to the request that created it. See [exceptions.py](https://github.com/requests/requests/blob/d8e236780981f28228441029c689a5e8dccdd71b/requests/exceptions.py#L22-L24). To make it more explicit, we could take a request object in our constructor as well.